### DISCUSSION TITLE
How to debug and optimize memory usage?

### DISCUSSION THREAD

**Discussion by HuakunShen** [tauri-apps/tauri#11553] [Q&A]
![image](https://github.com/user-attachments/assets/9b5b906d-cf64-4f03-ac98-28ac2e25314b)

My app is consuming more memory than expected and having memory leak problem.

What are each of the processes? 
Is `tauri://localhost` the rust tauri core process?
Is the second one the browser window process?

**Comment by FabianLars** [comment]
`tauri://localhost` is the main browser process (similar to a tab process in your browser).
the second one should be the rust process.
the other 2 are obviously the webview's/browser's sub-processes for gpu and network.

**Reply by HuakunShen** [reply]
Do you know any method to debug `tauri://localhost` memory usage? 

I looked at the timeline but the memory tab is always empty
![image](https://github.com/user-attachments/assets/ab162ee8-6eb6-4953-973e-d36accb16276)

**Reply by FabianLars** [reply]
no sorry. also that timeline is empty on my end too but i didn't find anything about that on the internet :/
